app/uk/gov/hmrc/hec/models/AuditEvent.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.models
+
+import play.api.libs.json.{JsObject, Json, OWrites}
+import uk.gov.hmrc.hec.models.hecTaxCheck.HECTaxCheck
+
+sealed trait AuditEvent {
+
+  val auditType: String
+
+  val transactionName: String
+
+}
+
+object AuditEvent {
+
+  final case class TaxCheckSuccess(
+    taxCheck: HECTaxCheck,
+    operatorDetails: Option[StrideOperatorDetails]
+  ) extends AuditEvent {
+    override val auditType: String = "TaxCheckSuccess"
+
+    override val transactionName: String = "tax-check-success"
+
+  }
+
+  object TaxCheckSuccess {
```

Comment:
Are we not creating any event on Tax check failure(no match /expired)?

Reply:
oh ok, please ignore..this is about saving tax check in db

Reply:
But this name "TaxCheckSuccess" lok a isleading . can we write something like TaxCheckStoreSuccess or something?

Reply:
I made it mirror what we initially agreed as the audit name with the CIP team - the idea was it described a tax check journey which was successful. This may change to something else like `TaxCheckResult` in the near future when we finalise it soon but we can just change it when it gets finalised I think.